Add <libc-pointer-arith.h> to simplify backports.

Introduced upstream as part of this commit:

commit 9090848d0607e93fb08a1d68d9f263846ee33f02
Author: Zack Weinberg <zackw@panix.com>
Date:   Sun Feb 26 20:17:52 2017 -0500

    Narrowing the visibility of libc-internal.h even further.

diff --git a/include/libc-pointer-arith.h b/include/libc-pointer-arith.h
new file mode 100644
index 0000000000000000..5876a41e8e5fcc15
--- /dev/null
+++ b/include/libc-pointer-arith.h
@@ -0,0 +1 @@
+#include <libc-internal.h>
